src/daemon/daemon.cpp

```diff
@@ -1790,3 +1912,33 @@ void mp::Daemon::start_mount(const VirtualMachine::UPtr& vm, const std::string&
 
     mount_threads[name][target_path] = std::move(sshfs_mount);
 }
+
+grpc::Status mp::Daemon::reboot_vm(VirtualMachine& vm, const VMSpecs& specs)
```

Comment:
You should be able to drop `const VMSpecs& specs` here since you can also get the vm's ssh_username via `vm.ssh_username()`.